api/pkg/imagebuilder/config.go

```diff
@@ -26,6 +26,8 @@ type Config struct {
 	// Path to sub folder which is intended to build instead of using root folder
 	ContextSubPath string
 	// Dockerfile Path within the build context
+	BaseImage cfg.BaseImageConfig
```

Comment:
Oh why do we have this config here and in `api/config.go` ? 🤔

Reply:
In cmd/setup.go we transform the config.ImageBuilderConfig to pkg.imagebuilder.Config: https://github.com/caraml-dev/merlin/blob/main/api/cmd/api/setup.go#L101-L141

When initializing imageBuilder object in pkg.imagebuilder (e.g. [NewModelServiceImageBuilder](https://github.com/caraml-dev/merlin/blob/main/api/pkg/imagebuilder/model_service_imagebuilder.go#L27)), we should be able to initialize it using its own internally-available Config struct, not external config.